### ISSUE TITLE
Reinstating emails that are currently not being sent

### ISSUE THREAD

**Issue by adishy** [UM-USElab/gradecraft-development#4359]
### Expected Behavior
The following emails should be sent:
*  `activation_needed_course_creation_email` (Sent when a new external user is created (when an instructor creates a trial account), i.e. when the user visits /users/new_external and creates a new account. Sent in controllers/users_controller.rb > create_external. This email was already being sent in the local environment but not in staging, could be a problem with email settings on staging.) (**Currently not being sent**)
* `app_resources_email` (Sent when a new external user is activated (when an instructor creates a trial account). The email is sent after the user visits the link in the activation email and clicks the activate account button. Sent in controllers/users_controller.rb > activated_external) (**Currently being sent**)
* `umich_resources_email` (**Currently not being sent**)
* `learning_objective_achieved`(**Currently being sent**)
*  `welcome_email` (**Currently not being sent**)
*  `new_submission` (Sends an email notification to instructors when a student submits an assignment. Added as a course wide setting in a separate PR, more details provided [here](https://github.com/UM-USElab/gradecraft-development/pull/4327)) (**Currently not being sent**)
* `revised_submission` (Sends an email notification to instructors when a student updates a submission. Added as a course wide setting in a separate PR, more details provided [here](https://github.com/UM-USElab/gradecraft-development/pull/4327)) (**Currently not being sent**)
*  `submissions_export_failure` (**Currently not being sent**)
*  `course_analytics_exports_mailer emails` (should be reinstated post analytics change) (**Currently not being sent**)
     * `export_failure` (**Currently not being sent**)
     *  `export_success`  (**Currently not being sent**)

### Actual Behavior
* Some of these emails are not being sent as the mailer for that email is not being called
* Some of these emails such as the course analytics export failure and export success can be sent only after the features are enabled in the interface.
* Some of the mailers for these emails such as the welcome email are being called in a background job such as in creates_new_user.rb and are not being sent
* Some of these emails are being sent locally but are not being sent on staging
* Testing is required to see which emails need reinstating and if the emails are being triggered at the correct place in the interface.
* `new_submission` and `revised_submission` is currently a course wide setting that sends emails to all instructors in the course when a student submits or resubmits an assignment and making this a instructor specific settings should be considered before this email is reinstated.

**Comment by maskib** [comment]
@adishy @ebarroso7 What's the status on this one? My understanding was that this had been largely addressed, but I wanted to confirm that with you all!
